### PR TITLE
ModuleLauncher uses ModuleResolver

### PR DESCRIPTION
**Pull request by markfisher**

### CODE REVIEW

File: spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java [hunk 1]
```diff
@@ -71,36 +79,38 @@ public static void main(String[] args) throws Exception {
...
+			this.artifactId = tokens[1];
+			this.version = tokens[2];
 			this.args = args;
 		}
 
 		@Override
 		public void run() {
 			try {
-				JarFileArchive jarFileArchive = new JarFileArchive(new File(directory, module));
+				Resource resource = moduleResolver.resolve(groupId, artifactId, version, "exec", "jar");
```
Comment (dturanski) [inline spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java]: The args are in a different order in https://github.com/spring-cloud/spring-cloud-stream/pull/47
 moduleResolver.resolve(groupId, artifactId,  "jar", "exec", version);

File: spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java [hunk 2]
```diff
@@ -71,36 +79,38 @@ public static void main(String[] args) throws Exception {
...
 
 		private final String[] args;
 
-		ModuleLaunchTask(File directory, String module, String[] args) {
-			this.directory = directory;
-			this.module = module;
+		ModuleLaunchTask(ModuleResolver moduleResolver, String module, String[] args) {
+			this.moduleResolver = moduleResolver;
+			String[] tokens = module.split(":");
+			Assert.isTrue(tokens.length == 3, "required module format is: 'groupId:artifactId:version'");
```
Comment (dturanski) [inline spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java]: Any possibility of the module string being group:artifact:extension:classifier:version or  group:artifact:extension:version?

File: spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java (hunk 1 above)
Comment (markfisher) [inline spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java]: yes I need to update this now... will resubmit

File: spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java (hunk 2 above)
Comment (markfisher) [inline spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java]: yes, need to update this

Comment (dturanski) [inline spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java]: Also starting to getting into shared code with ModuleCoordinates:  e.g.
       
       ModuleCordinates mc = ModuleCordinates.parse(module)
       moduleResolver.resolve(mc.getGroupId(), mc.getArtifactId(), mc.getExtension, mc.getClassifier(), mc.getVersion);

Comment (markfisher) [inline spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java]: We should avoid sharing that code IMO (for one thing, the ModuleLauncher will ultimately have different ideas for defaults, a bit more opinionated to support OOTB modules as the defaults, with the default group being "org.springframework.cloud.steam" etc). But I did copy the basic parsing code over in the update (2nd commit of this PR).

**Comment by dturanski** [comment]
LGTM; merged

**Pull request merged**